### PR TITLE
feedback: thuang-2366-feedback

### PR DESCRIPTION
**Pull request by tihuan**
#### Reviewers
**Functional:** 

**Readability:** 

---

## Changes
- add
- remove
- modify

### CODE REVIEW

File: client/src/actions/annotation.ts [hunk 1]
```diff
@@ -4,21 +4,20 @@ Action creators for user annotation
...
-  async (dispatch: any, getState: any) => {
+  async (dispatch: AppDispatch, getState: GetState): Promise<void> => {
     /*
-  Add a new user-created category to the obs annotations.
-
-  Arguments:
-    newCategoryName - string name for the category.
-    categoryToDuplicate - obs category to use for initial values, or null.
-  */
+     * Add a new user-created category to the obs annotations.
```
Comment (bkmartinjr) [inline client/src/actions/annotation.ts:16]: If you reformat comments, please convert to tsdoc style:
* comment goes before the function definition, not inside
* it starts with a `/**` not `/*`
* add `@params`
etc.

**Review by bkmartinjr — APPROVED** [review]
LGTM

Just a nit over tsdoc comment style

**Comment by tihuan** [comment]
Ooh that's right! Just updated to use the right format 🙆‍♂️ Thanks, Bruce!

**Comment by tihuan** [comment]
Asked for review again just in case lol Thank you!

File: client/src/actions/annotation.ts [hunk 2]
```diff
@@ -1,24 +1,22 @@
...
 import pako from "pako";
 import * as globals from "../globals";
+import { AppDispatch, GetState } from "../reducers";
 import { MatrixFBS, AnnotationsHelpers } from "../util/stateManager";
 
 const { isUserAnnotation } = AnnotationsHelpers;
 
+/**
+ * Add a new user-created category to the obs annotations.
+ * @arg newCategoryName - string name for the category.
```
Comment (bkmartinjr) [inline client/src/actions/annotation.ts:14]: should be `@param` and not `@arg`

See https://tsdoc.org/pages/tags/param/

Comment (bkmartinjr) [reply client/src/actions/annotation.ts:14]: Also see https://tsdoc.org/pages/tags/returns/

**Review by bkmartinjr — APPROVED** [review]
just a comment format nit, otherwise LGTM

File: client/src/actions/annotation.ts (hunk 2 above)
Comment (tihuan) [reply client/src/actions/annotation.ts]: Ahh only jsdoc has `@arg` alias

Comment (tihuan) [reply client/src/actions/annotation.ts]: This function returns `Promise<void>`, so I suppose we don't need `@returns`?

Comment (bkmartinjr) [reply client/src/actions/annotation.ts]: It is certainly optional, but my _personal opinion_ is that being explicit is very helpful - so I bias towards always specifying `@returns`.

Comment (tihuan) [reply client/src/actions/annotation.ts]: Ahh yeah that sounds good 👍 Just added `@returns An empty promise.`!

**Pull request merged**